### PR TITLE
Within and closeTo matchers for BigNumber

### LINKED ISSUE #347: Support additional chai matchers for BigNumber
It would be nice to have additional matchers that already work for plain numbers:

- `closeTo`
- `within`

Then a test like this:

```js
    it("Should return the correct task price before submission timeout", async () => {
      const expectedPrice = taskMinPrice.add(taskMaxPrice.sub(taskMinPrice).mul(secondsPassed).div(submissionTimeout));
      const delta = expectedPrice.div(100);

      const actualPrice = await contract.getTaskPrice(taskId, task);

      expect(actualPrice.sub(expectedPrice).abs()).to.be.lt(delta, "Invalid task price");
    });
```

could be written like:

```js
    it("Should return the correct task price before submission timeout", async () => {
      const expectedPrice = taskMinPrice.add(taskMaxPrice.sub(taskMinPrice).mul(secondsPassed).div(submissionTimeout));
      const delta = expectedPrice.div(100);

      const actualPrice = await contract.getTaskPrice(taskId, task);

      expect(actualPrice).to.be.closeTo(expectedPrice, delta, "Invalid task price");
    });
```

### PR DESCRIPTION
**Pull request by p-sad**
Closes #347

### CODE REVIEW

File: waffle-chai/test/matchers/bigNumber.test.ts [hunk 1]
```diff
@@ -122,4 +122,25 @@ describe('UNIT: BigNumber matchers', () => {
...
+
+  describe('closeTo', () => {
+    it('.to.be.closeTo', () => {
+      expect(BigNumber.from(100)).to.be.closeTo(BigNumber.from(101), 10);
+    });
+
+    it('.not.to.be.closeTo', () => {
+      expect(BigNumber.from(100)).not.to.be.closeTo(BigNumber.from(111), 10);
+    });
+  });
```
Comment (marekkirejczyk) [inline waffle-chai/test/matchers/bigNumber.test.ts:145]: Missing failing test cases.

File: waffle-chai/src/matchers/bigNumber.ts [hunk 2]
```diff
@@ -55,3 +59,49 @@ function overwriteBigNumberFunction(
     }
   };
 }
+
+function overrideWithin(utils: Chai.ChaiUtils) {
+  return (_super: (...args: any[]) => any) =>
+    overwriteBigNumberWithin(_super, utils);
```
Comment (marekkirejczyk) [inline waffle-chai/src/matchers/bigNumber.ts:65]: Could be inlined.

File: waffle-chai/src/matchers/bigNumber.ts [hunk 3]
```diff
@@ -24,6 +24,10 @@ export function supportBigNumber(
     'lte',
     override('lte', 'less than or equal', utils)
   );
+
```
Comment (marekkirejczyk) [inline waffle-chai/src/matchers/bigNumber.ts:27]: Also would be good to add documentation.

**Comment by rzadp** [comment]
Note: Closes https://github.com/EthWorks/Waffle/issues/347

**Comment by rzadp** [comment]
@p-sad Please change description to use [closing keywords](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Review by marekkirejczyk — APPROVED** [review]

**Pull request merged**